Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -392,5 +394,44 @@ private void btnChangeShell_Click(object sender, RoutedEventArgs e)
             if (LogoffChoice.HasValue && LogoffChoice.Value)
                 NativeMethods.Logoff();
         }
+
+        private void checkRunAtLogOn()
+        {
+            if (Shell.IsCairoUserShell.Equals(true))
+            {
+                chkRunAtLogOn.Visibility = Visibility.Hidden;
+            }
+
+            RegistryKey rKey = Registry.CurrentUser.CreateSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run");
+            List<string> rKeyValueNames = rKey.GetValueNames().ToList();
+
+            if (rKeyValueNames.Contains("Cairo"))
+            {
+                chkRunAtLogOn.IsChecked = true;
+            }
+            else
+            {
+                chkRunAtLogOn.IsChecked = false;
+            }
+        }
+
+        private void chkRunAtLogOn_Click(object sender, RoutedEventArgs e)
+        {
+            RegistryKey rKey = Registry.CurrentUser.CreateSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run");
+            var chkBox = (System.Windows.Controls.CheckBox)sender;
+
+            if (chkBox.IsChecked.Equals(false))
+            {
+                //Delete SubKey
+                rKey.DeleteValue("Cairo");
+            }
+            else
+            {
+                string exePath = System.Reflection.Assembly.GetExecutingAssembly().Location;
+                //Write SubKey
+                rKey.SetValue("Cairo", exePath);
```

Comment:
The installer sets the name as "CairoShell". This should be made consistent with that.